packages/core/src/components/forms/numericInput.tsx

```diff
@@ -107,6 +119,8 @@ export class NumericInput extends AbstractComponent<HTMLInputProps & INumericInp
         buttonPosition: Position.RIGHT,
         majorStepSize: 10,
         minorStepSize: 0.1,
+        selectAllOnFocus: false,
+        selectAllOnIncrement: false,
```

Comment:
+1 on consistency, therefore I think it should be `false`. If necessary, follow up with a PR that changes this default behavior to `true` across the board